src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AliasSymbolKey.cs

```diff
@@ -18,10 +18,16 @@ public static void Create(IAliasSymbol symbol, SymbolKeyWriter visitor)
                 visitor.WriteString(FirstOrDefault(symbol.DeclaringSyntaxReferences)?.SyntaxTree.FilePath ?? "");
             }
 
-            public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
+            public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string failureReason)
             {
                 var name = reader.ReadString();
-                var targetResolution = reader.ReadSymbolKey();
+                var targetResolution = reader.ReadSymbolKey(out var targetFailureReason);
+                if (targetFailureReason == null)
+                {
+                    failureReason = $"({nameof(AliasSymbolKey)} {nameof(targetResolution)} failed -> {targetFailureReason})";
```

Comment:
general pattern is:  `(this type failed resolving this piece of data -> due to this underlying cause)`.  The parens are there for nesting to make all the nested reasons something that can be grokked.